source/upgrade.txt

```diff
@@ -0,0 +1,101 @@
+.. _cpp-upgrade:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: compatibility, backwards compatibility
+
+Overview
+--------
+
+This page describes the changes you must make to your application
+when you upgrade to a new version of the {+driver-short+}.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the {+mdb-server+} versions
+  your application connects to and the C++ version your
+  application runs on. For version compatibility information, see the
+  :ref:`{+driver-short+} Compatibility <cpp-compatibility-tables>`
+  page.
+- Address any breaking changes between the driver version
+  your application is using and your planned upgrade version as detailed in the
+  :ref:`Breaking Changes <cpp-breaking-changes>` section.
+
+
+.. tip::
+
+   To minimize the number of changes your application requires when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <cpp-stable-api>`.
+
+.. _cpp-breaking-changes:
+
+Breaking Changes
+----------------
+
+.. note::
+
+   The {+driver-short+} doesn't ensure API or ABI stability across patch releases
+   within a minor version.
```

Comment:
```suggestion
   {+driver-short+} follows Semantic Versioning. Inadvertent API breaking changes may be reverted in a patch release within reason. The {+driver-short+} may break build system or package compatibility in any release. See: https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/api-abi-versioning/api-versioning/
```

I do not believe this is strictly correct. Although we follow SemVer and _may_ use "best judgment" to reverting an inadvertent API breaking change in a patch release (to preserve stability with the major version number), we do not expect this to ever be necessary. API breaking changes _should_ always constitute an API major version number bump. Recommend referencing/linking the [API Versioning](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/api-abi-versioning/api-versioning/) page for more details.

ABI stability is not guaranteed for _any_ release so far, as we do not yet have a stable ABI release.

If the intent was to document potential breaking changes which include _build system_ or _packaging_ compatibility (which seems to be the case given examples of "breaking changes" below), recommend explicitly stating so as suggested. See [Build System Compatibility](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/api-abi-versioning/api-versioning/#build-system-compatibility) for reference: we explicitly do _not_ consider properties of the build system to be part of our API compatibility contract unless it directly affects the API we _do_ support stability for.

Poking @kevinAlbs for confirmation.

Reply:
Suggested wording matches my expectation. I consider an API break in a minor release a bug. That bug may be fixed in a patch release.

Reply:
Appreciate the clarification!